### PR TITLE
strides enhancements and improvements 

### PR DESCRIPTION
**Pull request by notjustmat**
extends strides to 8 finger drum style pattern recorders, and 8 encoder recorders that send midi cc's.

Also fixes a few small bugs.

### CODE REVIEW

File: scripts/justmat/strides.lua [hunk 1]
```diff
@@ -284,142 +403,201 @@ end
...
-    slow_down_time()
-    slow_down_playback()
-  elseif n == 3 and z == 1 then
-    speed_up_time()
-    speed_up_playback()
+  -- key1 is alt
+  if n == 1 then alt_k = z end
+  -- alt + key3 changes modes
+  if alt_k == 1 and n == 3 and z == 1 then
+    if mode == 0 then
```
Comment (pq) [inline scripts/justmat/strides.lua]: FWIW, there's a kind of cheeky way to do ternaries in lua taking advantage of the quirky semantics of `and`.  For example you _could_ I think do something like this:

```lua
mode = mode == 0 and 1 or 0
```

(not that I'm necessarily suggesting it!)

Comment (pq) [reply scripts/justmat/strides.lua]: http://lua-users.org/wiki/ExpressionsTutorial

Comment (notjustmat) [reply scripts/justmat/strides.lua]: @pq this "not" suggestion, lol, just clicked for me! the look of ``mode = mode ==`` was throwing me. There may be a few places that could be done this way... 

Is there any benefit to using this method over whatever it is i'm doing currently?

Comment (pq) [reply scripts/justmat/strides.lua]: I'm not sure there are benefits.  The fact that the look of it throws you (and likely others) suggests on the contrary it'd be a liability!

**Pull request merged**